tests/e2e/system_test/test_graceful_nodes_shutdown.py

```diff
@@ -0,0 +1,419 @@
+        )
+        eb_restored_pvc = snapshot_restore_factory(
+            snapshot_obj=self.eb_snap_obj,
+            volume_mode=self.snap_obj.parent_volume_mode,
+        )
+        efs_restored_pvc = snapshot_restore_factory(
+            snapshot_obj=self.efs_snap_obj,
+            volume_mode=self.snap_obj.parent_volume_mode,
+        )
+
+        self.validate_pvc_expansion(
+            ne_restored_pvc, eb_restored_pvc, efs_restored_pvc, pvc_size_new=25
+        )
+
+    def validate_pvc_expansion(
+        self, ne_restored_pvc, eb_restored_pvc, efs_restored_pvc, pvc_size_new=25
+    ):
+        """
+        expand size of PVC and verify the expansion
+
+        Args:
+            pvc_size_new (int): Size of PVC(in Gb) to expand
+        """
+
+        for pvc_obj in ne_restored_pvc + eb_restored_pvc + efs_restored_pvc:
+            logger.info(f"Expanding size of PVC {pvc_obj.name} to {pvc_size_new}G")
+            pvc_obj.resize_pvc(pvc_size_new, True)
+
+    def validate_data_integrity(self):
+        """
+        Verifies the md5sum values of files are OK
+        Raises: If there is a mismatch in md5sum value or None
+        """
+        logger.info("Compare bucket object after reboot")
+        for idx, bucket_name in enumerate(self.bucket_names_list):
+            after_bucket_object_set = {
+                obj.key
+                for obj in self.mcg_obj.s3_list_all_objects_in_bucket(bucket_name)
+            }
+            assert (
+                self.before_bucket_object_list[idx] == after_bucket_object_set
+            ), f"Data integrity failed for S3 bucket {bucket_name}"
+
+        logger.info("Compare PVCs MD5Sum after reboot")
+        assert self.ne_pvc_orig_md5_sum == pod.cal_md5sum(
+            self.ne_pvc_pod_obj, self.ne_file_name
+        ), (
+            f"Data integrity failed for file '{self.ne_file_name}' "
+            f"on non encrypted pvc {self.ne_pvc_obj.name} on pod {self.ne_pvc_pod_obj.name}"
+        )
+
+        assert self.eb_pvc_orig_md5_sum == pod.cal_md5sum(
+            self.eb_pvc_pod_obj, self.eb_file_name
+        ), (
+            f"Data integrity failed for file '{self.eb_file_name}' "
+            f"on encrypted block pvc {self.eb_pvc_obj.name} on pod {self.eb_pvc_pod_obj.name}"
+        )
+
+        assert self.efs_pvc_orig_md5_sum == pod.cal_md5sum(
+            self.efs_pvc_pod_obj, self.efs_file_name
+        ), (
+            f"Data integrity failed for file '{self.efs_file_name}' "
+            f"on encrypted fs pvc {self.efs_pvc_obj.name} on pod {self.efs_pvc_pod_obj.name}"
+        )
+
+    def validate_ocp_workload_continues(self):
+        """
+        Verify ocp workload continues after reboot
+        """
+        if not check_if_monitoring_stack_exists():
+            assert "monitoring workload not started after reboot"
+        if not check_if_registry_stack_exists():
+            assert "registry workload not started after reboot"
```

Comment:
the workload exists does not mean that the workload is running fine without any issues after reboot

Reply:
Ok. what additional steps you suggest to check?